src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -560,7 +560,7 @@ public async Task StopAsync(CancellationToken cancellationToken)
             State = ScriptHostState.Stopped;
         }
 
-        public async Task RestartHostAsync(CancellationToken cancellationToken)
+        public async Task RestartHostAsync(string reason, CancellationToken cancellationToken)
```

Comment:
Because you're now wanting to ensure there's always a restart reason passed, perhaps you should add parameter validation to ensure the string is not null or empty

Reply:
Added validation.